plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/product-collection-content.tsx

```diff
@@ -75,7 +91,15 @@ const ProductCollectionContent = (
 				},
 				...( attributes as Partial< ProductCollectionAttributes > ),
 				queryId,
+				// If initialPreviewState is provided, use it, otherwise use default preview state.
+				...( !! attributes.collection && {
+					previewState:
+						initialPreviewState ||
+						( DEFAULT_ATTRIBUTES.previewState as PreviewState ),
```

Comment:
Please do not cast unless **absolutely** necessary (and in that case comment why). 

I checked, this one is redundant anyway, but you could be hiding type errors. (and don't forget to remove the import when removing the cast 😃  )

Reply:
Oh also this is wrong. DEFAULT_ATTRIBUTES does not contain this property. TS doesn't catch that because the type of DEFAULT_ATTRIBUTES is wrong.

It should be something like Pick<ProductCollectionAttributes, 'the', 'keys', 'i', 'want'> or Omit<ProductCollectionAttributes, 'exclude', 'these', 'keys'>

Reply:
> Please do not cast unless absolutely necessary (and in that case comment why).

Thanks for the reminder! I tend to default to casting when quickly coding for POCs, then forget to revise it when productionizing 🤦‍♂️. I'll make it a point to be more mindful about this from the start—it’s tough to spot later on.

> I checked, this one is redundant anyway, but you could be hiding type errors. (and don't forget to remove the import when removing the cast 😃 )

You're right, it was redundant. I've removed it 🤝.



> Oh also this is wrong. DEFAULT_ATTRIBUTES does not contain this property. TS doesn't catch that because the type of DEFAULT_ATTRIBUTES is wrong.
> 
> It should be something like Pick<ProductCollectionAttributes, 'the', 'keys', 'i', 'want'> or Omit<ProductCollectionAttributes, 'exclude', 'these', 'keys'>

Good point! I've made the changes and am now using Pick.